connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -2704,6 +2704,11 @@ public void onRevoked(String leader, Collection<String> connectors, Collection<C
             }
         }
 
+        @Override
+        public Stage onPollTimeoutExpiry() {
```

Comment:
This method name does not make sense. The name would indicate that this method is used to inform the callee (the listener, this implementation) that the poll timeout happened. The actual behavior and signature is a getter for the current stage.

I don't think it makes sense for the listener to require the caller to keep a stage, so i think the implementation should change to follow the method name.

Reply:
Yeah it seemed a bit weird to me as well but i thought that the log line for worker poll timeout needs to be added in `WorkerCoordinator` only. I have changed it to have it now being printed in the `WorkerRebalanceListener`. Let me know if it makes sense and that was something you had in mind.